src/ServiceInsight.Tests/SearchViewModelTests.cs

```diff
@@ -21,11 +22,15 @@ public SearchViewModelTests()
             var argParser = Substitute.For<CommandLineArgParser>();
             var settingProvider = Substitute.For<ISettingsProvider>();
 
+            // ReSharper disable once UnusedVariable
+            // required for async command
+            var dispatcher = Dispatcher.CurrentDispatcher;
```

Comment:
From https://msdn.microsoft.com/en-us/library/system.windows.threading.dispatcher.currentdispatcher(v=vs.110).aspx

> Gets the Dispatcher for the thread currently executing and creates a new Dispatcher if one is not already associated with the thread.

@danielmarbach will add more context.

Reply:
The SearchViewModel uses a ReactiveCommand that triggers the execute changed over Rx. Due to the context switch with async the RxObservable needs to be executed back on the UI thread. This is a strick here because accessing the CurrentDispatcher creates one that is associated with the current thread if there is not already one available thus allowing to execute the execute changed on a dispatcher.